src/tests/auth/test_auth.py

```diff
@@ -59,35 +61,40 @@ def test_get_roles(self, mock_extract_roles):
             delattr(mock_request, 'headers')
             self.assertEqual(_get_roles(), [])
 
-    @patch('gobstuf.auth.routes._get_roles')
-    def test_get_role(self, mock_get_roles):
-        mock_get_roles.return_value = []
-        self.assertIsNone(_get_role())
+    def test_get_role(self):
+        none_cases = [[], ['some ignored role'], ['brp_r']]
+        for case in none_cases:
+            self.assertIsNone(_get_role_fp(case))
 
-        mock_get_roles.return_value = ['some ignored role']
-        self.assertIsNone(_get_role())
+        roles = ['some ignored role', 'fp_the_role', 'brp_r']
+        self.assertEqual('fp_the_role', _get_role_fp(roles))
 
-        mock_get_roles.return_value = ['some ignored role', 'fp_the_role']
-        self.assertEqual('fp_the_role', _get_role())
+        roles = ['some ignored role', 'fp_the_role', 'fp_other_role', 'brp_r']
+        self.assertEqual('fp_the_role', _get_role_fp(roles))
```

Comment:
_ implies private, testing private functions is not best practice as they are implementation details which are hidden to users of that code.